### DISCUSSION TITLE
Prisma Throwing Error Cannot Connect To Database

### DISCUSSION THREAD

**Discussion by realswikarrr** [prisma/prisma#14667] [Q&A]
### Bug description

Hey guys so the issue is even though i am connected locally to the database prisma keeping throwing me a error saying 

```
Invalid `prisma.user.findUnique()` invocation:
Can't reach database server at `localhost`:`3309
```

```
generator client {
  provider = "prisma-client-js"
  previewFeatures = ["referentialIntegrity"]
  binaryTargets = ["native","darwin","debian-openssl-1.1.x"]
}

datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
  shadowDatabaseUrl = env("SHADOW_URL")
  referentialIntegrity = "prisma"
}
```

The url in env 

`DATABASE_URL="mysql://localhost:3309/shareu"`

I tried adding connection timeout at the end of the url but the result is same. It was working a day before with same and nothing is changed after that and now i am getting this error

Here's a screenshot of the error

![image](https://user-images.githubusercontent.com/80502023/182523800-d10538a6-ce88-4a58-a1e2-fe91d8f826ab.png)

### Prisma information

```
generator client {
  provider = "prisma-client-js"
  previewFeatures = ["referentialIntegrity"]
  binaryTargets = ["native","darwin","debian-openssl-1.1.x"]
}

datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
  shadowDatabaseUrl = env("SHADOW_URL")
  referentialIntegrity = "prisma"
}

model User {
  
  id String @id @default(cuid())

  createdAt DateTime @default(now())

  name String

  email String

  image String

  links Links[]

}

model Links {

  id String @id @default(cuid())

  createdAt DateTime @default(now())

  text String

  image String

  docs String

  user User @relation(fields: [userId], references: [id])

  userId String 
}
```

### Environment & setup

- OS: Manjaro Linux
- Database: MySQL
- Node.js version: 18.7.0

### Prisma Version

```
^4.1.1
```

**Comment by janpio** [comment]
Generally I would advise you to add a username and password to the connection string, but your statement that this used to work before and now does not any more invalidates that suggestion.

**Reply by realswikarrr** [reply]
well i fixed it by changing the branch

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
